source/base/function_signed_distance.cc

```diff
@@ -166,6 +166,42 @@ namespace Functions
 
 
 
+    template <int dim>
+    Tensor<1, dim>
+    Ellipsoid<dim>::gradient(const Point<dim> & point,
+                             const unsigned int component) const
+    {
+      AssertIndexRange(component, this->n_components);
+      (void)component;
+
+      const Point<dim> point_in_centered_cosy =
+        Point<dim>(compute_closest_point_ellipse(point) - center);
+
+      Tensor<1, dim> grad =
+        compute_analyical_normal_vector_on_ellipse(point_in_centered_cosy);
+      return grad / grad.norm();
+    }
+
+
+
+    template <int dim>
+    SymmetricTensor<2, dim>
+    Ellipsoid<dim>::hessian(const Point<dim> &,
+                            const unsigned int component) const
+    {
+      AssertIndexRange(component, this->n_components);
+      (void)component;
+
+      SymmetricTensor<2, dim> hess = unit_symmetric_tensor<dim>();
+
+      hess[0][0] = 2 / radii[0] * radii[0];
+      hess[1][1] = 2 / radii[1] * radii[1];
```

Comment:
This only works for 2d functions. But it shouldn't be difficult to write this via a `for` loop in a dimension independent way.

Separately, though, the computation is not correct. You want this:
```suggestion
      hess[0][0] = 2 / (radii[0] * radii[0]);
      hess[1][1] = 2 / (radii[1] * radii[1]);
```

Reply:
Playing around with my code today showed that this is not robust enough to estimate function bounds correctly for all cases. I suggest to use the hessian of a sphere with radius `(a + b) / 2` (see last commit).

Reply:
If anybody has a better idea how to compute a better approximation or even the analytic Hessian, suggestions are welcome!